### ISSUE TITLE
Missing hook for read of 'this'.

### ISSUE THREAD

**Issue by mtorp** [Haiyang-Sun/nodeprof.js#34]
In Jalangi the read of `this` triggers the read hook. It seems like no hook is triggered in nodeprof. Are there any plans on adding this feature to nodeprof as well?

**Comment by Haiyang-Sun** [comment]
@mtorp thanks for the feedback. The bug needs some Graal.js update. Please track this PR #36 and I will merge it when necessary changes are updated. For the moment, the patch below in Graal.js could fix the problem
```
diff --git a/graal-js/src/com.oracle.truffle.js/src/com/oracle/truffle/js/runtime/JSFrameUtil.java b/graal-js/src/com.oracle.truffle.js/src/com/oracle/truffle/js/runtime/JSFrameUtil.java
index 11dc8b7062..ea934e2e21 100644
--- a/graal-js/src/com.oracle.truffle.js/src/com/oracle/truffle/js/runtime/JSFrameUtil.java
+++ b/graal-js/src/com.oracle.truffle.js/src/com/oracle/truffle/js/runtime/JSFrameUtil.java
@@ -110,7 +110,7 @@ public final class JSFrameUtil {
             String name = (String) frameSlot.getIdentifier();
             if (name.startsWith(":")) {
                 return true;
-            } else if (name.startsWith("<") && name.endsWith(">")) {
+            } else if (name.startsWith("<") && name.endsWith(">") && !name.equals("<this>")) {
                 return true;
             }
             return false;
```

@eleinadani  ```this``` variable is currently checked as an internal variable in Graal.js while it shouldn't. It's different from other internal variables and shouldn't be hidden from the instrumentation. 
- https://github.com/graalvm/graaljs/blob/4d40a6ef2b13bcf6bcf4d5e0b31c3a62a4fa416e/graal-js/src/com.oracle.truffle.js/src/com/oracle/truffle/js/runtime/JSFrameUtil.java#L107-L119

**Comment by eleinadani** [comment]
I agree this sounds like a Graal.js issue. @Haiyang-Sun your patch above is wrong: the `<this>` variable is  internal: we use a frame slot to store a reference to `this`, but it should _not_ be exposed. I think we can fix this issue in a different way: do you have a small unit test to reproduce?

**Comment by Haiyang-Sun** [comment]
```
    @Test
    public void readThis() {
        evalWithTags("(()=>{return this;})()", new Class[]{JSTags.ReadVariableExpressionTag.class});

        enter(ReadVariableExpressionTag.class, (e, r) -> {
            assertAttribute(e, NAME, "<this>");
        }).exit();
    }
```

**Comment by alexjordan** [comment]
Fixed in #36
